src/async_api/async_api_internal/tokio/io_ops/nonblocking_io.rs

```diff
@@ -0,0 +1,101 @@
+use super::FileDescriptorRawHandle;
+use libc::O_NONBLOCK;
+use libc::{self, c_int, c_void, size_t};
+use log::error;
+use std::sync::Arc;
+use std::{io, time::Duration};
+use tokio::io::unix::AsyncFd;
+
+#[derive(Debug, Clone)]
+pub struct SubChannel {
+    fd: Arc<AsyncFd<FileDescriptorRawHandle>>,
```

Comment:
It might be better to impl `AsRawFd` for `Channel` and `ChannelSender`, that way you can implement this in terms of them.  This would become `AsyncFd<Channel>`.  It would remove a lot of the duplication here and you would inherit the safety of the channel implementation.